### PR TITLE
refactored application/applet descriptor elements

### PR DESCRIPTION
**Pull request by AndreasEhret**
- refactor/supplement descriptor elements according to JSR spec and coding conventions
- adapted naming according to JSR spec
- introduced ApplicationType according to JSR spec
- refactored JavaFX parameter as ApplicationType
- introduced EntryPoint abstraction

### CODE REVIEW

File: core/src/main/java/net/sourceforge/jnlp/util/AppletUtils.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AppletUtils {
+    private final static Logger LOG = LoggerFactory.getLogger(AppletUtils.class);
+
+    /**
+     * @return the width of the Applet in pixels
+     */
+    public static int getFixedWidth(int width, final Map<String, String> parameters) {
```
Comment (hendrikebbers) [inline core/src/main/java/net/sourceforge/jnlp/util/AppletUtils.java]: final for width

File: core/src/main/java/net/sourceforge/jnlp/util/AppletUtils.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                return nww;
+            }
+        }
+        return width;
+    }
+
+    /**
+     * @return the height of the Applet in pixels
+     */
+    public static int getFixedHeight(int height, final Map<String, String> parameters) {
```
[the remark this answers is not in the document]
Comment (AndreasEhret) [reply core/src/main/java/net/sourceforge/jnlp/util/AppletUtils.java]: done

File: core/src/main/java/net/sourceforge/jnlp/util/AppletUtils.java (hunk 1 above)
Comment (AndreasEhret) [reply core/src/main/java/net/sourceforge/jnlp/util/AppletUtils.java]: done

File: core/src/main/java/net/sourceforge/jnlp/runtime/JNLPClassLoader.java [hunk 3]
```diff
@@ -935,9 +935,9 @@ void checkForMain(List<JARDesc> jars) throws LaunchException {
 
         // Check launch info
         if (mainClass == null) {
-            LaunchDesc launchDesc = file.getLaunchInfo();
-            if (launchDesc != null) {
-                mainClass = launchDesc.getMainClass();
+            EntryPoint entryPoint = file.getEntryPointDesc();
```
[the remark this answers is not in the document]
Comment (AndreasEhret) [reply core/src/main/java/net/sourceforge/jnlp/runtime/JNLPClassLoader.java]: done

File: core/src/main/java/net/sourceforge/jnlp/Parser.java [hunk 4]
```diff
@@ -731,19 +731,19 @@ public LaunchDesc getLauncher(Node parent) throws ParseException {
 
         Node child = parent.getFirstChild();
         while (child != null) {
-            String name = child.getNodeName().getName();
+            final String name = child.getNodeName().getName();
 
             if ("applet-desc".equals(name)) {
```
[the remark this answers is not in the document]
Comment (AndreasEhret) [reply core/src/main/java/net/sourceforge/jnlp/Parser.java]: can we leave this to one of my next upcoming pull requests in pipeline where I do the refactoring of the Parser class?

File: core/src/main/java/net/sourceforge/jnlp/Parser.java [hunk 5]
```diff
@@ -752,20 +752,20 @@ public LaunchDesc getLauncher(Node parent) throws ParseException {
...
      */
-    private AppletDesc getApplet(Node node) throws ParseException {
-        String name = getRequiredAttribute(node, "name", R("PUnknownApplet"));
-        String main = getMainClass(node, true);
-        URL docbase = getURL(node, "documentbase", base);
-        Map<String, String> paramMap = new HashMap<>();
+    private AppletDesc getApplet(final Node node) throws ParseException {
+        final String name = getRequiredAttribute(node, "name", R("PUnknownApplet"));
+        final String main = getMainClass(node, true);
+        final URL docbase = getURL(node, "documentbase", base);
```
[the remark this answers is not in the document]
Comment (AndreasEhret) [reply core/src/main/java/net/sourceforge/jnlp/Parser.java]: can we leave this to one of my next upcoming pull requests in pipeline where I do the refactoring of the Parser class?

File: core/src/main/java/net/sourceforge/jnlp/Parser.java [hunk 6]
```diff
@@ -793,24 +793,26 @@ private AppletDesc getApplet(Node node) throws ParseException {
...
-    private ApplicationDesc getApplication(Node node, boolean isFx) throws ParseException {
+    private ApplicationDesc getApplication(final ApplicationType applicationType, final Node node) throws ParseException {
         String main = getMainClass(node, false);
         List<String> argsList = new ArrayList<>();
 
         // if (main == null)
         //   only ok if can be found in main jar file (can't check here but make a note)
         // read parameters
-        Node args[] = getChildNodes(node, "argument");
+        final Node args[] = getChildNodes(node, "argument");
```
[the remark this answers is not in the document]
Comment (AndreasEhret) [reply core/src/main/java/net/sourceforge/jnlp/Parser.java]: can we leave this to one of my next upcoming pull requests in pipeline where I do the refactoring of the Parser class?

**Pull request merged**